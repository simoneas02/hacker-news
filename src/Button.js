import React, { Component } from 'react';

class Button extends Component {
    render () {
        const { 
            onClick, 
            className = '', 
            children 
        } = this.props;

        return (
            <button
                onClick={ onClick }
                className={ className }
                type='button'
            >
                { children }
            </button>
        )
    }
}

export default Button;