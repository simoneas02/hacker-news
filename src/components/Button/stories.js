import React from 'react';
import { storiesOf } from '@storybook/react';

import Button from '.';

storiesOf('Button', module).add("I'm App", () => <h1>Hello App</h1>);
