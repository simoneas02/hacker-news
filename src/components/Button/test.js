import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Button from '.';

describe('Button', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Button />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
