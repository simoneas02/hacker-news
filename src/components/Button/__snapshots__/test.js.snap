// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has a valid snapshot 1`] = `
<button
  className=""
  onClick={undefined}
  type="button"
/>
`;
